### PR TITLE
Feature/6742 vaccination certificate/6940 json validation

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds a json schema validation to the health certificate.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7081

### CODE REVIEW

**Review by whiskey — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -28,6 +37,24 @@
...
+      },
+      {
+        "package": "Spectre",
+        "repositoryURL": "https://github.com/kylef/Spectre.git",
+        "state": {
+          "branch": null,
+          "revision": "f79d4ecbf8bc4e1579fbd86c3e1d652fb6876c53",
+          "version": "0.9.2"
+        }
+      },
```
Comment (pascal-brause) [inline src/xcode/ENA.xcworkspace/xcshareddata/swiftpm/Package.resolved:57]: are these two packages in the JSONSchema package included? Do we need to mention them in the licenses, too?

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess.swift [hunk 2]
```diff
@@ -74,20 +74,38 @@ struct CertificateAccess {
...
     // MARK: - Private
 
+    private func validateSchema(of certificate: CBOR) -> Result<Void, CertificateDecodingError> {
+        guard case let CBOR.map(certificateMap) = certificate,
+              let schemaURL = Bundle.module.url(forResource: "CertificateSchema", withExtension: "json"),
+              let schemaData = FileManager.default.contents(atPath: schemaURL.path),
+              let schemaDict = try? JSONSerialization.jsonObject(with: schemaData) as? [String: Any],
+              let validationResult = try? JSONSchema.validate(certificateMap, schema: schemaDict),
+              case .valid = validationResult else {
+
```
Comment (pascal-brause) [inline src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess.swift:102]: perhaps add a log here?

File: src/xcode/ENA.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA.xcworkspace/xcshareddata/swiftpm/Package.resolved:57]: No, I checked that with Marcus, its not needed.

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/CertificateAccess.swift:102]: Logging is not possible at the moment. We need to discuss a solution in the DEV-Camp. I have added an entry for that.

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**